### PR TITLE
[BAU] table_nullable added to validation

### PR DESCRIPTION
**Pull request by cyrusdobbs**
### Change description
adds the concept of nullable tables

- [X] Unit tests and other appropriate tests added or updated
- [X] README and other documentation has been updated / added (if needed)
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### CODE REVIEW

File: core/validation/validate.py [hunk 1]
```diff
@@ -68,12 +68,14 @@ def validate_workbook(workbook: dict[str, pd.DataFrame], schema: dict) -> list[v
...
         (validate_enums, "enums"),
-        (validate_nullable, "non-nullable"),
+        (validate_nullable, "non-nullable"),  # TODO: add tests for this
     )
 
     validation_failures = []
     for sheet_name in workbook.keys():
-        if workbook[sheet_name].empty:
+        # if the table is empty and not defined as nullable, then raise an Empty Sheet Failure
+        # TODO: add tests for this
```
Comment (nsnape) [inline core/validation/validate.py:77]: could we bang a quick test in while we're here, looks like it should be easy? Means we don't add that to the tech-debt mountain 😄

**Review by nsnape — CHANGES_REQUESTED** [review]

File: core/validation/validate.py (hunk 1 above)
Comment (cyrusdobbs) [reply core/validation/validate.py:77]: yeah sure

Comment (cyrusdobbs) [reply core/validation/validate.py]: Added

File: core/validation/schema.py [hunk 2]
```diff
@@ -53,10 +53,15 @@ def parse_schema(schema: dict) -> Optional[dict]:
                 assert all(isinstance(value, str) for value in enum_values)
                 sheet_schema["enums"][column] = set(enum_values)  # cast to a set
 
-            nullable = sheet_schema.get("nullable", [])
-            for column in nullable:
+            # TODO: add tests for this
```
Comment (nsnape) [inline core/validation/schema.py:56]: do these comments need removing now?

File: core/validation/validate.py [hunk 3]
```diff
@@ -68,12 +68,13 @@ def validate_workbook(workbook: dict[str, pd.DataFrame], schema: dict) -> list[v
         (validate_uniques, "uniques"),
         (validate_foreign_keys, "foreign_keys"),
         (validate_enums, "enums"),
-        (validate_nullable, "non-nullable"),
+        (validate_nullable, "non-nullable"),  # TODO: add tests for this
```
Comment (nsnape) [inline core/validation/validate.py:71]: tested now?

**Review by nsnape — APPROVED** [review]

**Pull request merged**